### PR TITLE
Auto translate

### PR DESCRIPTION
**Pull request by ivandrofly**
- Move json processing to thread-pool
- Reuse http-client

Related to https://github.com/SubtitleEdit/subtitleedit/pull/8549

Note: If you want to still allow the user to change the value from combobox while ollama client is waiting 
for the models request let me know

### CODE REVIEW

File: src/ui/Forms/Translate/AutoTranslate.cs [hunk 1]
```diff
@@ -1455,47 +1455,59 @@ private async void UpdateLocalModelsToolStripMenuItem_Click(object sender, Event
...
             }
         }
+
+        private HttpClient _httpClient;
+
+        private HttpClient GetOllamaClient()
+        {
+            return _httpClient ?? (_httpClient = new HttpClient
+            {
+                Timeout = TimeSpan.FromSeconds(3)
```
Comment (ivandrofly) [inline src/ui/Forms/Translate/AutoTranslate.cs:1507]: Keep the timer? @niksedk

File: src/ui/Forms/Translate/AutoTranslate.cs [hunk 2]
```diff
@@ -1172,7 +1172,7 @@ private async void nikseComboBoxEngine_SelectedIndexChanged(object sender, Event
             if (GetCurrentEngine().GetType() == typeof(OllamaTranslate))
             {
                 nikseComboBoxEngine.Enabled = false;
-                await DownloadOllamaModelsAsync().ConfigureAwait(true);
```
Comment (ivandrofly) [inline src/ui/Forms/Translate/AutoTranslate.cs:1175]: Redundant

File: src/ui/Forms/Translate/AutoTranslate.cs [hunk 3]
```diff
@@ -1457,45 +1457,60 @@ private async void UpdateLocalModelsToolStripMenuItem_Click(object sender, Event
 
         private async Task DownloadOllamaModelsAsync()
         {
-            using (var httpClient = new HttpClient())
+            try
             {
-                try
+                var models = await GetModelsAsync(nikseComboBoxUrl.Text.Replace("generate", "tags")).ConfigureAwait(true);
```
Comment (ivandrofly) [inline src/ui/Forms/Translate/AutoTranslate.cs:1462]: Now the UI thread is only adding the models to the combobox

File: src/ui/Forms/Translate/AutoTranslate.cs [hunk 4]
```diff
@@ -1457,45 +1457,60 @@ private async void UpdateLocalModelsToolStripMenuItem_Click(object sender, Event
...
-                    {
-                        return;
-                    }
+                    FillOllamaModels(models.ToArray());
+                }
+            }
+            catch (Exception exception)
+            {
+#if DEBUG
+                MessageBox.Show(exception.Message);
```
Comment (ivandrofly) [inline src/ui/Forms/Translate/AutoTranslate.cs:1471]: Display full exception message when in debug mode

**Comment by niksedk** [comment]
The idea is that while using the engine combo box SE will try to refresh the ollama models once, but only using a shorter time - and not blocking the user from selecting another engine, and no errors.

If selecting refresh models context menu on the models combo box, then a longer timeout + display error.

**Comment by ivandrofly** [comment]
> The idea is that while using the engine combo box SE will try to refresh the ollama models once, but only using a shorter time - and not blocking the user from selecting another engine, and no errors.
> 
> If selecting refresh models context menu on the models combo box, then a longer timeout + display error.

Updated

**Pull request merged**